tests/trainers/test_i_jepa_trainer.py

```diff
@@ -0,0 +1,167 @@
+assert IMAGE_SIZE % PATCH_SIZE == 0
+N_PATCHES = IMAGE_SIZE // PATCH_SIZE
+
+# model params
+ENCODER_EMBEDDING_DIM = 48
+ENCODER_NUM_HEADS = 6
+assert ENCODER_EMBEDDING_DIM % ENCODER_NUM_HEADS == 0
+PREDICTOR_EMBEDDING_DIM = 24
+PREDICTOR_NUM_HEADS = 3
+assert PREDICTOR_EMBEDDING_DIM % PREDICTOR_NUM_HEADS == 0
+
+
+class TestIJEPATrainer:
+    @pytest.fixture
+    def partial_dataloader(self):
+        partial_dataloader = partial(DataLoader, batch_size=2, shuffle=True)
+        return partial_dataloader
+
+    @pytest.fixture
+    def partial_optimizer(self):
+        partial_optimizer = partial(AdamW, lr=1e-4, weight_decay=0.04)  # based on the original paper's initial params
+        return partial_optimizer
+
+    @pytest.fixture
+    def i_jepa_mask_collator(self):
+        # based on the original paper settings
+        i_jepa_mask_collator = IJEPAMultiBlockMaskCollator(
+            input_size=IMAGE_SIZE,
+            patch_size=PATCH_SIZE,
+            encoder_mask_scale=(0.85, 1.0),
+            predictor_mask_scale=(0.15, 0.2),
+            aspect_ratio=(0.75, 1.5),
+            n_masks_for_context_encoder=1,
+            n_masks_for_predictor=4,
+            min_keep=10,
+        )
+        return i_jepa_mask_collator
+
+    @pytest.fixture
+    def i_jepa_encoder(self):
+        return IJEPAEncoder(
+            img_size=IMAGE_SIZE,
+            patch_size=PATCH_SIZE,
+            embed_dim=ENCODER_EMBEDDING_DIM,
+            depth=3,
+            num_heads=ENCODER_NUM_HEADS,
+            mlp_ratio=4.0,
+        )
+
+    @pytest.fixture
+    def i_jepa_predictor(self):
+        return IJEPAPredictor(
+            n_patches=N_PATCHES,
+            context_encoder_embed_dim=ENCODER_EMBEDDING_DIM,
+            predictor_embed_dim=PREDICTOR_EMBEDDING_DIM,
+            depth=3,
+            num_heads=PREDICTOR_NUM_HEADS,
+        )
+
+    @pytest.fixture
+    def image_step_data(self) -> StepData:
+        d = StepData()
+        d[DataKeys.OBSERVATION] = torch.randn(3, IMAGE_SIZE, IMAGE_SIZE)
+        return d
+
+    @pytest.fixture
+    def image_buffer_dict(self, image_step_data: StepData) -> DataCollectorsDict:
+        d = DataCollectorsDict.from_data_buffers(
+            **{BufferNames.IMAGE: RandomDataBuffer.reconstructable_init(32, [DataKeys.OBSERVATION])}
+        )
+
+        d.collect(image_step_data)
+        d.collect(image_step_data)
+        d.collect(image_step_data)
+        d.collect(image_step_data)
+        return d
+
+    @pytest.fixture
+    def model_wrappers_dict(
+        self, device: torch.device, i_jepa_encoder: IJEPAEncoder, i_jepa_predictor: IJEPAPredictor
+    ) -> ModelWrappersDict:
+        d = ModelWrappersDict(
+            {
+                "i_jepa_context_encoder": ModelWrapper(i_jepa_encoder, device, False),
+                "i_jepa_predictor": ModelWrapper(i_jepa_predictor, device, False),
+                "i_jepa_target_encoder": ModelWrapper(copy.deepcopy(i_jepa_encoder), device, False),
```

Comment:
ModelNamesを使用しよう。また、target encoderは agentの推論スレッドでも用いられるので`has_inference`を`True`にしよう。
```suggestion
                ModelNames.I_JEPA_CONTEXT_ENCODER: ModelWrapper(i_jepa_encoder, device, False),
                ModelNames.I_JEPA_PREDICTOR: ModelWrapper(i_jepa_predictor, device, False),
                ModelNames.I_JEPA_TARGET_ENCODER: ModelWrapper(copy.deepcopy(i_jepa_encoder), device, True),
```

Reply:
https://github.com/MLShukai/ami-vconf24/pull/47/commits/313328dcabae403ee6e5357837e2eaae3a439d7f で変更